plugin/hack.vim

```diff
@@ -107,11 +107,34 @@ function! hack#toggle()
   endif
 endfunction
 
+function! hack#format(from, to)
+  if !executable(g:hack#hh_format)
+    echo 'g:hack#hh_format not executable'
+  endif
+
+  if &modified
+    echo 'Error[hack]: buffer has unsaved changes'
+    return
+  endif
+  let frombyte = line2byte(a:from)
+
+  let tobyte = line2byte(a:to) + strlen(getline(a:to))
+
+  execute a:from.','.a:to.' ! '.g:hack#hh_format.
+    \ ' --from '.frombyte.' --to '.tobyte.
+    \ ' --root '.g:hack#root.' '.expand('%:p')
+  let tmp = g:hack#enable
+  " Disable auto checking
+  let g:hack#enable = 0
+  silent write
+  let g:hack#enable = tmp
```

Comment:
What does the `execute` command you're invoking do, exactly?  Is it replacing the selected lines with the hh_format output?  If that'll be reflected in the buffer, it'd be better not to do a write here—let the user see the changes before they commit to them.